### DISCUSSION TITLE
Serratus Explorer querying experience

### DISCUSSION THREAD

**Discussion by victorlin** [serratus-bio/serratus.io#120] [Ideas]
#112 requires some under-the-hood rewriting and also some UI decisions. Here's a blueprint:

<img width="422" alt="serratus-explorer" src="https://user-images.githubusercontent.com/13424970/111017455-afe56b80-8381-11eb-9d2b-861f2dd54e9b.png">

Each _Search type_ selection will yield different options for _Match type_:

- Nucleotide: family/genbank/run
- Protein: family/protein/sequence
- RdRP: phylum/family/rdrp

Main points of discussion:

- Are _Search type_ and _Match type_ the right labels here? I was thinking _Match level_ to reflect the hierarchal relation amongst the options, but "SRA Run" isn't really a level - rather a summary of matches from all levels.
- Do we want to re-visit the idea of having a "priming" feature and/or one dropdown per match type? https://github.com/serratus-bio/serratus.io/issues/79#issuecomment-678723122

**Comment by ababaian** [comment]
An alternative would be to "tab" out through the different search types. I think it will be easy to get confused "what kind of data" am I looking at, and if we keep a distinct visual identity for each search type (blues, reds, purples) then you'll know where you are.

![serratusio2](https://user-images.githubusercontent.com/9220907/111020243-46615f00-8379-11eb-9a97-5509953722ef.png)

The examples do that "Priming" very well in my opinion, we would just have to include more variety of them (i.e. example of a GenBank Search too) and then for each "tab" we would have a different set of examples in the priming.

**Reply by victorlin** [reply]
Nice idea. Proposed URL examples:

- `serratus.io/explorer/nucleotide?family=Coronaviridae`
- `serratus.io/explorer/rdrp?family=Coronaviridae`
- `serratus.io/explorer/protein?protein=rdrp`

**Reply by ababaian** [reply]
Let's focus on `rdrp` and `nucleotide` first, the protein data was a pilot search I think it's less critical to get it operational soon.  But the idea of generalizing the results is what is important since we will add more "experiments" as time goes forward.

We can shorten it to `rdrp` and `nt` for the URL

**Comment by ababaian** [comment]
With respect to the labels: `Search Type` I think works. Maybe a more accurate description for the next selection would be `Query by`, as in how are you querying the data: "SRA Run", "GenBank", "Family" all make sense in this respect

**Reply by victorlin** [reply]
Not sure about this one - since `Search` and `Query` are synonymous, having both `Search Type` and `Query by` could be confusing to an inexperienced user.

I'm also trying to avoid confusion in the code - currently using `queryType` for family/genbank/run and it's already a bit confusing since there exists other "querying" options (score/identity - these are termed `filters` internally). And the number of options will only grow when we add geo+time filters.

**Reply by victorlin** [reply]
`Search Type` / `Search Level` ?

**Reply by victorlin** [reply]
I went ahead with `Search Type` / `Search Level` in the code: https://github.com/serratus-bio/serratus.io/pull/118/commits/c7a09110508b8661a6c4cfe7d1f308c32655cefc

**Reply by ababaian** [reply]
sounds good :D
